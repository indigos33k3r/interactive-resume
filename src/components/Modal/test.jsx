import React from 'react'
import {shallow} from 'enzyme'

describe('Modal', () => {
  it('renders <Modal />', () => {
    const component = render()

    expect(component).toMatchSnapshot()
  })

  context('when <ReactModal /> #prop.onHide was triggered', () => {
    it('sets `state.showModal` to `false`', () => {
      const component = render()
      component.find('Modal').prop('onHide')()
      component.update()

      expect(component.state().showModal).toBe(false)
    })
  })

  context('when <ReactModal /> #prop.onExited was triggered', () => {
    it('calls #props.onExited', () => {
      const onExited = jest.fn()
      const component = render({onExited})
      component.find('Modal').prop('onExited')()

      expect(onExited).toHaveBeenCalledTimes(1)
    })
  })

  context('when <ReactModal /> #prop.onEnter was triggered', () => {
    it('calls #props.onEnter', () => {
      const onEnter = jest.fn()
      const component = render({onEnter})
      component.find('Modal').prop('onEnter')()

      expect(onEnter).toHaveBeenCalledTimes(1)
    })
  })

  function render(props = {}) {
    const defaultProps = {
      children: 'Some children text',
      modalClass: 'SomeModalClass',
      onExited: () => {},
      onEnter: () => {}
    }
    const Modal = require('.').default

    return shallow(<Modal {...defaultProps} {...props} />)
  }
})
