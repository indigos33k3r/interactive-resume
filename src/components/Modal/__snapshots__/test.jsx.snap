// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders <Modal /> 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogClassName="SomeModalClass"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onEnter={[Function]}
  onExited={[Function]}
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={true}
    closeLabel=""
  />
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    Some children text
  </ModalBody>
</Modal>
`;
